flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/expressionDsl.scala

```diff
@@ -355,190 +155,10 @@ trait ImplicitExpressionOperations {
     * @param ifFalse expression to be evaluated if condition does not hold
     */
   def ?(ifTrue: Expression, ifFalse: Expression): Expression =
-    unresolvedCall(IF, expr, ifTrue, ifFalse)
+    Expressions.ifThenElse(expr, ifTrue, ifFalse).toExpr
```

Comment:
should we offer an infix notation for `ifThenElse`? I think it would be nice to fluently evaluate a boolean predicate.

Reply:
Any suggestions for a name?

Reply:
difficult, here are some inspirations: https://en.wikipedia.org/wiki/%3F:

How about "then" or "ifThenElse" again: `$("f").equalTo(12).then(12, null)`